cmd/snap-confine/mount-support.c

```diff
@@ -647,9 +647,6 @@ void sc_ensure_shared_snap_mount(void)
 
 static void sc_make_slave_mount_ns(void)
 {
-	if (unshare(CLONE_NEWNS) < 0) {
-		die("can not unshare mount namespace");
-	}
```

Comment:
Removing this from here makes this function a one line function, called from only one place (sc_setup_user_mounts). As such, I think I'd prefer this patch to move this:

```
	// In our new mount namespace, recursively change all mounts
	// to slave mode, so we see changes from the parent namespace
	// but don't propagate our own changes.
	sc_do_mount("none", "/", NULL, MS_REC | MS_SLAVE, NULL);
```

down into sc_setup_user_mounts() and then removing sc_make_slave_mount_ns() entirely.

Reply:
I agree but I wanted to keep it separate. In another branch I simplified this whole chunk. I was after a psychological effect where a patch that removes code without adding any code is easier to review.

I will open a follow up.